hypertrace-gradle-docker-java-application-plugin/src/main/resources/application-start-script.template.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+#
+# Copyright 2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+##############################################################################
+##
+##  ${applicationName} start up script for UN*X
+##
+##############################################################################
+
+set -e
+
+# Add default JVM options here. You can also use JAVA_OPTS to pass JVM options to this script.
+DEFAULT_JVM_OPTS=${defaultJvmOpts}
+
+# Escape application args
```

Comment:
for those curious. this approach is lifted from gradlew https://github.com/gradle/gradle/blob/c92004f2bb135faaf3beda1e39fb6e8aff6ca6cb/subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt#L175-L183

unsolicited 2p:

This isn't generally done like this in java, even outside docker, as it makes assumptions about the type of variables passed. When copy/pasting things into scripts probably it is better to have a specific use case, and cite the source in case that changes later.

For example, something like dancing around quote nesting and order like gradlew does is not guaranteed to be a problem we will have in our images, and it is more complex to explain why we need this vs a simpler alternative that assumes you don't need to re-process all things into a single quoted string.

IOTW That's why I suggested simply ordering  "$@" at the end, as if there was an edge case we actually use it would smoke it out, vs lift something complex without others knowing why or if we need it.

Reply:
That's fine, I can remove the arg escaping. The general approach I took was to start with the exact gradle behavior, and peel out the irrelevant stuff. I would usually argue it's a good thing to keep the `gradlew run` command used in development as similar as possible to the artifact's own entrypoint. That said, we don't really pass args anyway and like you mentioned, it's just another source of questions to a viewer.